File: 346. Moving Average from Data Stream.cpp
class MovingAverage {
public:
    /** Initialize your data structure here. */
    queue<int> myQueue;
    int maxSize;
    double currentSum;
    MovingAverage(int size) {
        maxSize = size;
        currentSum = 0;
    }

    double next(int val) {
        if(myQueue.size() == maxSize) {
            currentSum -= myQueue.front();
            myQueue.pop();
        }
        myQueue.push(val);
        currentSum += val;
        return currentSum / myQueue.size();
    }
};

/**
 * Your MovingAverage object will be instantiated and called as such:
 * MovingAverage obj = new MovingAverage(size);
 * double param_1 = obj.next(val);
 */

File: 346. Moving Average from Data Stream.java
class MovingAverage {

    private Queue<Integer> queue;
    private int maxSize;
    private double currentSum;
    /** Initialize your data structure here. */
    public MovingAverage(int size) {
        queue = new LinkedList<>();
        maxSize = size;
        currentSum = 0;
    }

    public double next(int val) {
        if(queue.size() == maxSize) {
            int headValue = queue.remove();
            currentSum -= headValue;
        }
        currentSum += val;
        queue.add(val);
        return currentSum / queue.size();
    }
}

/**
 * Your MovingAverage object will be instantiated and called as such:
 * MovingAverage obj = new MovingAverage(size);
 * double param_1 = obj.next(val);
 */
